dev/tests/integration/testsuite/Magento/Catalog/Model/Product/UrlTest.php

```diff
@@ -42,6 +42,34 @@ public function testGetUrlInStore()
         $this->assertStringEndsWith('simple-product.html', $this->_model->getUrlInStore($product));
     }
 
+    /**
+     * @magentoDataFixture Magento/Store/_files/second_store.php
+     * @magentoConfigFixture default_store web/unsecure/base_url http://sample.com/
+     * @magentoConfigFixture default_store web/unsecure/base_link_url http://sample.com/
+     * @magentoDataFixture Magento/Catalog/_files/products_in_different_stores.php
+     * @magentoAppArea adminhtml
+     */
+    public function testGetUrlInStoreWithSecondStore()
+    {
+        $repository = \Magento\TestFramework\Helper\Bootstrap::getObjectManager()->create(
+            \Magento\Catalog\Model\ProductRepository::class
+        );
+        /** @var \Magento\Store\Model\Store $store */
+        $store = \Magento\TestFramework\Helper\Bootstrap::getObjectManager()
+            ->create(\Magento\Store\Model\Store::class);
+        $store->load('fixturestore', 'code');
+        $product = $repository->get('simple_2');
+
+        $this->assertEquals(
+            'http://sample.com/index.php/simple-product-2.html',
+            $this->_model->getUrlInStore($product, ['_scope'=> 1, '_nosid' => true])
+        );
+        $this->assertNotEquals(
```

Comment:
The changes are non-trivial, so I'd prefer strict comparisons to expected result for both scopes (ideally organized with data provider)